src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -256,11 +256,14 @@ export class ConnectionWidget extends lifecycle.Disposable {
 	protected addCustomConnectionOptions(): void {
 		if (this._customOptions.length > 0) {
 			this._customOptionWidgets = [];
+			let trueInputValue = localize('boolean.true', 'True');
+			let falseInputValue = localize('boolean.false', 'False');
 			this._customOptions.forEach((option, i) => {
-				let customOptionsContainer = DialogHelper.appendRow(this._tableContainer, option.displayName, 'connection-label', 'connection-input', 'custom-connection-options');
+				let customOptionsContainer = DialogHelper.appendRow(this._tableContainer, option.displayName, 'connection-label', 'connection-input', 'custom-connection-options', false, option.description);
 				switch (option.valueType) {
 					case ServiceOptionType.boolean:
-						this._customOptionWidgets[i] = new SelectBox([localize('boolean.true', 'True'), localize('boolean.false', 'False')], option.defaultValue, this._contextViewService, customOptionsContainer, { ariaLabel: option.displayName });
+						let optionValue = (option.defaultValue.toString() === true.toString() || option.defaultValue === trueInputValue) ? trueInputValue : falseInputValue;
```

Comment:
defaultValue is already a string so why are you calling toString on it?

Although isn't the defaultValue type wrong in this case? You're setting it to a boolean so it should be a boolean type too right?

Reply:
DefaultValue can be boolean for boolean types as it is allowed to be set in package.json. The interface declares it as string as a generic type, so we cannot compare with boolean directly (as compiler complains), and we cannot compare it as string without converting defaultValue to string either (since its bool and comparison fails at runtime). 

I'll add a comment here for future reference, but the best way of comparison is by converting both to string in this case.

Reply:
If we allow boolean types then we should declare the type as `string | boolean` (and whatever other types we support). Not having it means that anyone using this type could easily end up with runtime errors - for example right now if someone called `option.defaultValue.toLowerCase()` when defaultValue was actually a boolean it would throw.

And this is something we should be enforcing when we read in the options from the package.json contribution to avoid unexpected breaking behavior if someone puts in an expected value for a defaultValue (or any of the other types). See https://github.com/Microsoft/azuredatastudio/blob/8a3d08f0dec1cdf47d7fc442f90fb2227b828375/src/vs/workbench/services/actions/common/menusExtensionPoint.ts#L646 for an example of how this could be done.

Reply:
I created https://github.com/microsoft/azuredatastudio/issues/21023 to track this issue for all contributions and I'll plan on following up on it later on when I have some free time - for this PR though at the very least you need to update the types to allow booleans as well (although if you wanted to go add the validation now for this contribution that'd be fantastic).

Reply:
I agree we need to atleast support `boolean | string | number` as all three types currently are being set in values and default values. Since ConnectionOptions interface is decoupled with package.json, it let this gap flow through. 

If you agree we can track it separately?